rules/Php82/Rector/New_/FilesystemIteratorSkipDotsRector.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php82\Rector\New_;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\BitwiseOr;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name\FullyQualified;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\Php82\Rector\New_\FilesystemIteratorSkipDots\FilesystemIteratorSkipDotsRectorTests
+ */
+class FilesystemIteratorSkipDotsRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Prior PHP 8.2 FilesystemIterator::SKIP_DOTS was always set and could not be removed, therefore FilesystemIterator::SKIP_DOTS is added in order to keep this behaviour.',
+            [new CodeSample(
+                'new \\FilesystemIterator(__DIR__, \\FilesystemIterator::KEY_AS_FILENAME);',
+                'new \\FilesystemIterator(__DIR__, \\FilesystemIterator::KEY_AS_FILENAME | \\FilesystemIterator::SKIP_DOTS);'
+            ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [New_::class];
+    }
+
+    /**
+     * @param New_ $node
+     */
+    public function refactor(Node $node): ?New_
+    {
+        if (! array_key_exists(1, $node->args)) {
+            return null;
+        }
+        $flags = $node->args[1]->value;
+        if ($this->isSkipDotsPresent($flags)) {
+            return null;
+        }
+        $skipDots = new ClassConstFetch(new FullyQualified('FilesystemIterator'), 'SKIP_DOTS');
+        $node->args[1] = new Arg(new BitwiseOr($flags, $skipDots));
+        return $node;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::FILESYSTEM_ITERATOR_SKIP_DOTS;
+    }
+
+    /**
+     * Is the constant {@see \FilesystemIterator::SKIP_DOTS} present?
+     */
+    private function isSkipDotsPresent(Expr $node): bool
```

Comment:
I prefer to verify the expr is ConstFetch or `BitwiseOr`, so verify the target ConstFetch exists in the `BitwiseOr`

Reply:
@samsonasik Hello, I have made the requested change using ClassConstFetch, but I also made a mistake while rebasing and now there's +100 files in this pull request :(
I'm thinking about closing this PR and create a new one, what's your opinion?

Reply:
I wrote a blog post about how to rebase https://samsonasik.wordpress.com/2015/09/16/practical-git-4-rebasing-conflicted-task-branch-against-primary-branch/

Reply:
Hello, I applied requested changes :+1:

Reply:
could you rebase latest main branch?

Reply:
Hello, I just rebased the branch